### PR TITLE
[LMLayer] Trie weighting

### PR DESCRIPTION
**Pull request by eddieantonio**
First draft of trie weighting. This adds both the necessary enhancements to let the trie model yield probabilities based on the word count, but also adds a small integration test of the trie model used in conjunction with the model compositor.

The next step will be to precompute the total weight of the trie in the model compiler, so it does not have to recomputed on the instantiation of the trie.

**EDIT**: assumes #1811 has been merged.

### CODE REVIEW

**Comment by keyman-server** [comment]
[OSK renders for Windows](https://build.palaso.org/repository/download/KeymanDesktop_TestPrRenderOnScreenKeyboards/152868:id/kvk/index.html)

File: common/predictive-text/worker/models/trie-model.ts [hunk 1]
```diff
@@ -323,16 +339,38 @@
...
+
   }
-  
+
+  /**
+   * O(n) lookup to determine the total amount of nodes in the trie, starting
+   * at the provided node. Don't use this if you want lookups to be fast.
+   * @param node
+   */
+  function countNodes(node: Node): number {
```
Comment (jahorton) [inline common/predictive-text/worker/models/trie-model.ts]: So, this adds up the weights from the leaves, correct?  Not the 'total amounts of nodes in the trie'?  I'm looking at example.str.trie_sencoten.model.js for comparison, and its weights are far higher than this comment would suggest.

Comment (eddieantonio) [reply common/predictive-text/worker/models/trie-model.ts]: Good catch! That's correct; this should say it counts the weights of the leaves, and not how many nodes there are themselves.

**Comment by eddieantonio** [comment]
I renamed the method to `sumWeights()` which is what it does. Note that this functionality should be present in the compiler, and should not need to be invoked during model load time. I will address this in a future PR.

**Pull request merged**